src/__tests__/keyboard/keyboardImplementation.ts

```diff
@@ -15,3 +15,197 @@ test('no character input if `altKey` or `ctrlKey` is pressed', () => {
   expect(eventWasFired('keypress')).toBe(false)
   expect(eventWasFired('input')).toBe(false)
 })
+
+describe('fire events', () => {
+  it('fires event with releasing key twice', () => {
+    const {element, getEventSnapshot, clearEventCalls} = setup(`<input/>`)
+
+    ;(element as HTMLInputElement).focus()
+    clearEventCalls()
+
+    userEvent.keyboard('{ArrowLeft>}{ArrowLeft}')
+
+    expect(getEventSnapshot()).toMatchInlineSnapshot(`
+Events fired on: input[value=""]
+
+input[value=""] - keydown: ArrowLeft (37)
+input[value=""] - select
+input[value=""] - keyup: ArrowLeft (37)
+input[value=""] - keydown: ArrowLeft (37)
+input[value=""] - select
+input[value=""] - keyup: ArrowLeft (37)
+`)
+  })
+
+  it('fires event without releasing key', () => {
+    const {element, getEventSnapshot, clearEventCalls} = setup(`<input/>`)
+
+    ;(element as HTMLInputElement).focus()
+    clearEventCalls()
+
+    userEvent.keyboard('{a>}')
+
+    expect(getEventSnapshot()).toMatchInlineSnapshot(`
+Events fired on: input[value="a"]
+
+input[value=""] - keydown: a (97)
+input[value=""] - keypress: a (97)
+input[value="a"] - input
+`)
+  })
+
+  it('fires event multiple times without releasing key', () => {
+    const {element, getEventSnapshot, clearEventCalls} = setup(`<input/>`)
+    ;(element as HTMLInputElement).focus()
+    clearEventCalls()
+
+    userEvent.keyboard('{a>2}')
+
+    expect(getEventSnapshot()).toMatchInlineSnapshot(`
+Events fired on: input[value="a"]
+
+input[value=""] - keydown: a (97)
+input[value=""] - keypress: a (97)
+input[value="a"] - input
+`)
+  })
+
+  it('fires event multiple times and releases key', () => {
+    const {element, getEventSnapshot, clearEventCalls} = setup(`<input/>`)
+    ;(element as HTMLInputElement).focus()
+    clearEventCalls()
+
+    userEvent.keyboard('{a>2}{/a}')
+
+    expect(getEventSnapshot()).toMatchInlineSnapshot(`
+Events fired on: input[value="aa"]
+
+input[value=""] - keydown: a (97)
+input[value=""] - keypress: a (97)
+input[value="a"] - input
+input[value="a"] - keyup: a (97)
+input[value="a"] - keydown: a (97)
+input[value="a"] - keypress: a (97)
+input[value="aa"] - input
+input[value="aa"] - keyup: a (97)
```

Comment:
See this log produced in the browser:
```
{"type":"keydown","key":"a","code":"KeyA"}
{"type":"keypress","key":"a","code":"KeyA"}
{"type":"input","data":"a","inputType":"insertText","value":"a"}
{"type":"keydown","key":"a","code":"KeyA"}
{"type":"keypress","key":"a","code":"KeyA"}
{"type":"input","data":"a","inputType":"insertText","value":"aa"}
{"type":"keydown","key":"a","code":"KeyA"}
{"type":"keypress","key":"a","code":"KeyA"}
{"type":"input","data":"a","inputType":"insertText","value":"aaa"}
{"type":"keyup","key":"a","code":"KeyA"}
```

You can use [this codesandbox](https://codesandbox.io/s/keyevent-vtdcc?file=/src/App.js) to test out events.